app/src/main/java/com/github/onlynotesswent/ui/common/Popups.kt

```diff
@@ -137,6 +149,108 @@ fun CreationDialog(
             }
       })
 }
+/**
+ * Displays a popup dialog to browse and navigate the file system.
+ *
+ * This popup allows users to view folders in a hierarchical structure. Users can navigate into
+ * subfolders, view folder contents, and return to the root folder.
+ *
+ * @param onDismiss Callback invoked when the popup is dismissed.
+ * @param folderViewModel The ViewModel that provides the folder data.
+ */
+@Composable
+fun FileSystemPopup(onDismiss: () -> Unit, folderViewModel: FolderViewModel) {
```

Comment:
Out of scope for this PR, but in a future PR we should make this pop up more general so that we can reuse for decks and flashcards

Reply:
Yes that is the goal of my task for this week.